### PR TITLE
Update spec

### PR DESCRIPTION
**Pull request by arthurdm**
- Updated static file location / format (META-INF with openapi[yml | yaml | json]
- Updated architecture diagram
- Updated annotation overview table
- Update annotation samples (just missing Schema and Tags)

### CODE REVIEW

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc [hunk 1]
```diff
@@ -118,25 +118,36 @@ of another native configuration framework) must use the prefix `mp.openapi.exten
...
+OpenAPI <<Annotations>>.  Using annotations means
 developers don't have to re-write the portions of the OpenAPI document that are
 already covered by the JAX-RS framework (e.g. the HTTP method of an operation).
 
-Another way to document your APIs is via <<Static OpenAPI files>>.  This
-is often adopted by enterprises that want to lock-in the contract of the API
-before any code is written.  In this case, we refer to the OpenAPI document as the
-"source of truth", by which the client and provider must abide.
+2. Take the initial output from `/openapi` as a starting point to document
+your APIs is via <<Static OpenAPI files>>.  It's worth mentioning that these static
```
Comment (EricWittmann) [inline spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: "to document your APIs is via ..." - maybe just drop the word "is"?

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc [hunk 2]
```diff
@@ -186,35 +227,225 @@ public Response findPetsByStatus(...) { ... }
...
+[source,Java]
+----
+@GET
+@Path("/{username}")
+@Operation(summary = "Get user by user name",
+        responses = {
+                @APIResponse(description = "The user",
+                        content = @Content(mediaType = "application/json",
+                                schema = @Schema(implementation = User.class))),
+                @AAPIResponse(responseCode = "400", description = "User not found")})
```
Comment (EricWittmann) [inline spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: The `AAPIResponse` has too many A's?

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc [hunk 3]
```diff
@@ -186,35 +227,225 @@ public Response findPetsByStatus(...) { ... }
...
+.Sample 2 - Operation with different responses
+[source,Java]
+----
+@GET
+@Path("/{username}")
+@Operation(summary = "Get user by user name",
+        responses = {
+                @APIResponse(description = "The user",
+                        content = @Content(mediaType = "application/json",
+                                schema = @Schema(implementation = User.class))),
```
Comment (EricWittmann) [inline spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: Inconsistent tabbing of the previous 3 lines - the `description` and `content` and `schema` attributes should line up.

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc [hunk 4]
```diff
@@ -186,35 +227,225 @@ public Response findPetsByStatus(...) { ... }
...
 ----
-
+@POST
+        @Path("/user")
+        @Operation(summary = "Create user",
+                description = "This can only be done by the logged in user.")
+        public Response methodWithRequestBodyAndTwoParameters(
+                @RequestBody(description = "Created user object", required = true,
+                        content = @Content(
+                                schema = @Schema(implementation = User.class))) User user,
```
Comment (EricWittmann) [inline spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: I think `description`, `content`, and `schema` should align.

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc [hunk 5]
```diff
@@ -186,35 +227,225 @@ public Response findPetsByStatus(...) { ... }
...
-.Sample 1
+
+.Sample 1 - Simple RequestBody
 [source,Java]
 ----
-
+@POST
+        @Path("/user")
+        @Operation(summary = "Create user",
+                description = "This can only be done by the logged in user.")
```
Comment (EricWittmann) [inline spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: I think `summary` and `description` should align.

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc [hunk 6]
```diff
@@ -186,35 +227,225 @@ public Response findPetsByStatus(...) { ... }
...
+      schema:
+      type: string
+    - name: code
+      in: query
+      schema:
+      type: string
+    requestBody:
+      description: Created user object
+      content:
+      '*/*':
```
Comment (EricWittmann) [inline spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: Does the entire `'*/*'` section need to be indented one more level?  I think it's a child of `content`, right?

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc (hunk 1 above)
Comment (arthurdm) [reply spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: yup

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc (hunk 3 above)
Comment (arthurdm) [reply spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: +1

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc (hunk 2 above)
Comment (arthurdm) [reply spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: +1

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc (hunk 5 above)
Comment (arthurdm) [reply spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: yes

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc (hunk 6 above)
Comment (arthurdm) [reply spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: right.  copy/paste error

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc (hunk 4 above)
Comment (arthurdm) [reply spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: +1

File: spec/src/main/asciidoc/microprofile-openapi-spec.adoc [hunk 7]
```diff
@@ -180,41 +221,231 @@ public Response findPetsByStatus(...) { ... }
...
+@POST
+        @Path("/user")
+        @Operation(summary = "Create user",
+                   description = "This can only be done by the logged in user.")
+        public Response methodWithRequestBodyAndTwoParameters(
+                @RequestBody(description = "Created user object", required = true,
+                             content = @Content(
+                             schema = @Schema(implementation = User.class))) User user,
+                @QueryParam("name") String name, @QueryParam("code") String code)
+                { ... }
```
Comment (EricWittmann) [inline spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: Indentation of this block is wrong - everything except `@POST` should be de-dented one level.  In other words, `@Path`, `@Operation`, and `public Response...` should all be at the same level as `@POST`.

Comment (arthurdm) [reply spec/src/main/asciidoc/microprofile-openapi-spec.adoc]: +1

**Pull request merged**